### PR TITLE
Deprecating the "Authoring markdown" section and adding config docs for the new legacy flag

### PR DESCRIPTION
**Pull request by tony-sull**
#### What kind of changes does this PR include?

Updating the docs to reflect the new "legacy" mode for Astro-flavored markdown in [PR #4016](https://github.com/withastro/astro/pull/4016)

#### Description

- Deprecates the "Authoring markdown" section with a note on how to re-enable the features until they are removed in a future Astro release (no timeline on that, I'd guess 2.0?)
- Adds configuration docs for the new `legacy.astroFlavoredMarkdown` flag

### CODE REVIEW

File: src/pages/en/guides/markdown-content.md [hunk 1]
```diff
@@ -172,6 +172,10 @@ You can also pass the `--drafts` flag when running `astro build` to build draft
 
 ## Authoring Markdown
 
+:::caution[Deprecated]
+Astro no longer supports components or JSX in markdown pages by default and may be removed in a future release. In the meantime, Astro config supports a [legacy flag](/en/reference/configuration-reference/#legacyastroflavoredmarkdown) that will enable these features in markdown pages until you are able to migrate to [`@astrojs/mdx`](/en/guides/integrations-guide/mdx/).
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:176]:
```suggestion
Astro no longer supports components or JSX in Markdown pages by default and may be removed in a future release. In the meantime, Astro config supports a [legacy flag](/en/reference/configuration-reference/#legacyastroflavoredmarkdown) that will enable these features in Markdown pages until you are able to migrate to [`@astrojs/mdx`](/en/guides/integrations-guide/mdx/).
```
We capitalize Markdown in the docs, so I've changed those.

Just crossed my mind: is it worth pointing out that there are two options: migrate to using MDX *or* traditional Markdown (i.e. remove your comonents/JSX)?  Might not be, but it's technically a possibility.

**Comment by sarah11918** [comment]
Thanks for this, @tony-sull !

Two related notes:

1. The `configuration-reference` page is auto-generated from `@types/astro.ts` in the main repo, so we don't make changes to it here on our end. Please make sure these changes are made in the astro repo, if you haven't already, and you can just discard them here.

3. Do you also want to add content at this time for the `migration` guide page? Or would you like that "on the stack"? (I believe we already have a note in an issue we're monitoring for Migration Guide content that it's on our radar to expect.)

**Comment by tony-sull** [comment]
🤦 I always forget the Configuration Reference is auto-generated, the docs project is too fancy for me! 😅 

Just removed the manual config reference updates, I'll go ahead and take care of the migration guide while I'm here!

File: src/pages/en/migrate.md [hunk 2]
```diff
@@ -36,6 +36,12 @@ const canonicalURL = Astro.canonicalURL;
 const canonicalURL = new URL(Astro.url.pathname, Astro.site);
 ```
 
+### Deprecated: Components and JSX in Markdown
+
+Astro no longer supports components or JSX expressions in Markdown pages by default. For long term support you should migrate to the [`@astrojs/mdx`]() integration.
```
Comment (sarah11918) [inline src/pages/en/migrate.md:41]:
```suggestion
Astro no longer supports components or JSX expressions in Markdown pages by default. For long-term support you should migrate to the [`@astrojs/mdx`](/en/guides/integrations-guide/mdx/) integration.
```
Just added the link in!

**Comment by sarah11918** [comment]
Note: this link check is going to fail until the changes from `astro.ts` are merged, and then make their way here. So, the PR is fine, but I'll hold off on this until the link to the new configuration flag exists!

**Comment by sarah11918** [comment]
This one now depends on https://github.com/withastro/astro/pull/4027 with updated content for the configuration-reference page.

Once that merges, and the next day the changes make it into docs, then we can double-check the anchor link for the new option, and this should good to go.

**Comment by sarah11918** [comment]
@tony-sull This looks like it passes the link checker now that the legacy flag exists on the configuration reference page, so should be good to merge!

**Pull request merged**